app/experimenter/nimbus-ui/README.md

```diff
@@ -75,6 +75,106 @@ export interface GetExperimentOverviews {
+  <Form.Text>This is the description.</Form.Text>
+  {errors["description"] && (
+    <Form.Control.Feedback type="invalid">
+      {errors["description"].message}
+    </Form.Control.Feedback>
+  )}
+</Form.Group>
+```
+
+### General errors
+
+General errors can occur when an operation fails altogether. This could be for any reason, such as when a network request fails, or a bad code path raises an exception. For these errors you'll use React Bootstrap's [`Alert`](https://react-bootstrap.github.io/components/alerts/) components.
+
+For example, if error occurs while saving a record:
+
+```tsx
+<Alert variant="warning">Sorry, there was a problem.</Alert>
+```
+
+_Note:_ displaying technical error messages to the user may not be very helpful, so it's generally advised to either provide a [generic message](./src/lib/constants.ts) or instructions on what to do.
+
+## Handling GraphQL operation errors
+
+As GQL queries and mutations are how we get data in and out of the app this is where a majority of our errors are going to occur. These should be handled in a consistent fashion as well.
+
+Both operations can result in a general error, such as a `NetworkError`, or GraphQL errors, which in most cases will be validation errors. Importantly, these two types of errors occur in different ways; general errors will throw or trigger the operations `onError` function, and GraphQL errors will be present in the result of the operation.
+
+At this time we're not requiring any errors to be reported to Sentry as uncaught client errors will bubble up to `AppErrorBoundary`, we don't want to report validation errors, and true GraphQL server errors are already reported on the server-side.
+
+An example of handling a `useMutation` operation:
+
+```tsx
+// Set up the Mutation
+const [updateSomething] = useMutation(UPDATE_RECORD_MUTATION);
+
+// Later, execute it.
+try {
+  const result = await updateSomething({
+    variables,
+  });
+
+  if (!result.data?.updateSomething) {
+    throw new Error("Update failed for an unknown reason");
+  }
+
+  const { message, record } = result.data.updateSomething;
+
+  // In our GraphQL response, the `data.message` could be "success"
+  // to indicate successful mutation, another error string, or an
+  // object containing field keys and an array of string errors.
+  if (message !== "success" && typeof message === "object") {
+    console.log("A GraphQL error occurred", message);
+    return;
+  }
+
+  console.log("Record updated", record);
+} catch (error) {
+  console.log("A general error occurred", error.message);
+}
+```
+
+An example of handling a `useQuery` operation:
+
+```tsx
+// Set up the query, and because it executes immediately, handle it
+// immediately by passing in an `onError` option. This is the same as
+// wrapping it in a try/catch, just a little more grouped together.
+const { data } = useQuery(GET_RECORD_QUERY, {
+  onError(error) {
+    console.log("A general error occurred", error.message);
```

Comment:
Did we want a component or something here to display errors with? Or maybe we'll just display at the top of the page somewhere.

Reply:
In an effort not to overkill this section I'm just demonstrating that "this is where you need to do something", and the `<Alert>` component is demonstrated in the section above, which is what you would need to use for something like this, but would involve setting up state and some other stuff to demonstrate here.

Reply:
Ohh I see, maybe ignore my last comment then, _this_ is where we'd use `<Alert>`